Update if...else statements to c.Asserts:  Issue #16756

zelahi opened the pull request:
Updated docker_cli_kill_test.go to change if...else statements to c.Asserts

Signed-off-by: Zuhayr Elahi elahi.zuhayr@gmail.com

aidanhs committed:
Update ambassador image, use the socat -t option

Signed-off-by: Aidan Hobson Sayers <aidanhs@cantab.net>

swernli committed:
Fixing hang in archive.CopyWithTar with invalid dst

Signed-off-by: Stefan J. Wernli <swernli@microsoft.com>

aidanhs committed:
Remove references to the docker-ut image

Signed-off-by: Aidan Hobson Sayers <aidanhs@cantab.net>

GabrielNicolasAvellaneda committed:
MINOR typo fix.

Signed-off-by: GabrielNicolasAvellaneda <avellaneda.gabriel@gmail.com>

mountkin commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -17,9 +18,8 @@ func (s *DockerSuite) TestKillContainer(c *check.C) {
-	if strings.Contains(out, cleanedContainerID) {
-		c.Fatal("killed container is still running")
-	}
+	c.Assert(out, checker.Contains, cleanedContainerID, check.Commentf("killed container is still running"))
```
should be `checker.Not(checker.Contains)`

mountkin commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -42,9 +42,11 @@ func (s *DockerSuite) TestKillDifferentUserContainer(c *check.C) {
+	processContainerID := strings.TrimSpace(out)
+
+	c.Assert(waitRun(processContainerID),check.IsNil)
+	c.Assert(out, checker.Contains, cleanedContainerID, check.Commentf("killed container is still running"))
```
should be `checker.Not(checker.Contains)`

zelahi commented:
Part of #16756

vdemeester commented:
@zelahi you need to `gofmt` the file : `gofmt -s -w  integration-cli/docker_cli_kill_test.go`.

zelahi commented:
@vdemeester Thanks for the tips!! I just ran it =)

vdemeester commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -57,9 +59,9 @@ func (s *DockerSuite) TestKillWithSignal(c *check.C) {
-		c.Fatal("Container should be in running state after SIGWINCH")
-	}
+
+	c.Assert(waitRun(running), check.IsNil)
```
I think this line is not needed and make the build fail.

vdemeester commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -42,9 +42,11 @@ func (s *DockerSuite) TestKillDifferentUserContainer(c *check.C) {
-	}
+	processContainerID := strings.TrimSpace(out)
+
+	c.Assert(waitRun(processContainerID), check.IsNil)
```
This line and the line above are not right (and not needed) and make the build fail.

zelahi replied:
okay, so should I remove the if..else and the c.Assert statements?  Or is it better to leave it as an if..else statement.  I could also rewrite c.Assert statement?  

If I am understanding the test correctly, it checks to see that the containers are killed properly.  And at these lines its making sue that the containers are not running anymore.  Is it expected that the container is supposed to still be running?

vdemeester replied:
> `waitRun` will wait for the specified container to be running, maximum 5 seconds

So in that case, the containers has been already kill, it will return an error.

Anyway, it wasn't done before, it does not need to be done after the change. No need to add extra checks, you just need `c.Assert(out, checker.Not(checker.Contains), cleanedContainerID, check.Commentf("killed container is still running"))` to replace the `if …` :wink:.

zelahi replied:
O awesome. Thanks for the tip @vdemeester .  I'll remove them =)

albers committed:
Bash completion for restart policy `unless-stopped`

Signed-off-by: Harald Albers <github@albersweb.de>

thaJeztah committed:
docs: fix storage driver options list

This fixes the indentation of the storage driver
options list.

Also wraps/reformats some examples to prevent
horizontal scrollbars on the rendered HTML

Fixes #17140

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Use of checkers on docker_cli_nat_test.go.

Signed-off-by: liaoqingwei <liaoqingwei@huawei.com>

Update docker_cli_kill_test.go to change if...else statements to c.Assert

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

Ran gofmt to format docker_cli_kill_test.go

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

modified test to remove c.Asserts which were causing failures and changed checker.contains to checker.equals

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

thaJeztah committed:
docs: fix storage driver options list

This fixes the indentation of the storage driver
options list.

Also wraps/reformats some examples to prevent
horizontal scrollbars on the rendered HTML

Fixes #17140

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Use of checkers on docker_cli_nat_test.go.

Signed-off-by: liaoqingwei <liaoqingwei@huawei.com>

Update docker_cli_kill_test.go to change if...else statements to c.Assert

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

Ran gofmt to format docker_cli_kill_test.go

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

modified test to remove c.Asserts which were causing failures and changed checker.contains to checker.equals

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

docs: fix storage driver options list

This fixes the indentation of the storage driver
options list.

Also wraps/reformats some examples to prevent
horizontal scrollbars on the rendered HTML

Fixes #17140

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Use of checkers on docker_cli_nat_test.go.

Signed-off-by: liaoqingwei <liaoqingwei@huawei.com>

Update docker_cli_kill_test.go to change if...else statements to c.Assert

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

Ran gofmt to format docker_cli_kill_test.go

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

modified test to remove c.Asserts which were causing failures and changed checker.contains to checker.equals

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

thaJeztah committed:
docs: fix storage driver options list

This fixes the indentation of the storage driver
options list.

Also wraps/reformats some examples to prevent
horizontal scrollbars on the rendered HTML

Fixes #17140

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Use of checkers on docker_cli_nat_test.go.

Signed-off-by: liaoqingwei <liaoqingwei@huawei.com>

Update docker_cli_kill_test.go to change if...else statements to c.Assert

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

add chec.Not around checker.Contains on lines 21 and 48

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

Ran gofmt to format docker_cli_kill_test.go

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

modified test to remove c.Asserts which were causing failures and changed checker.contains to checker.equals

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

mountkin commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -57,9 +56,8 @@ func (s *DockerSuite) TestKillWithSignal(c *check.C) {
-		c.Fatal("Container should be in running state after SIGWINCH")
-	}
+
+	c.Assert(running, checker.Contains, "true", check.Commentf("Container should be in running state after SIGWINCH"))
```
should be `checker.Equals`

mountkin commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -70,14 +68,10 @@ func (s *DockerSuite) TestKillWithInvalidSignal(c *check.C) {
-	if running != "true" {
-		c.Fatal("Container should be in running state after an invalid signal")
-	}
+	c.Assert(running, checker.Contains, "true", check.Commentf("Container should be in running state after an invalid signal"))
```
should be `checker.Equals`

mountkin commented on integration-cli/docker_cli_kill_test.go:
```diff
@@ -85,14 +79,11 @@ func (s *DockerSuite) TestKillWithInvalidSignal(c *check.C) {
-	if running != "true" {
-		c.Fatal("Container should be in running state after an invalid signal")
-	}
+	c.Assert(running, checker.Contains, "true", check.Commentf("Container should be in running state after an invalid signal"))
```
should be `checker.Equals`

zelahi replied:
@mountkin , thanks for your help. =)

albers committed:
bash completion: support for dm.use_deferred_* options

Signed-off-by: Harald Albers <github@albersweb.de>

zelahi commented:
Is there a way to debug janky, or the Jenkins?  I looked at the log and it said that it failed to destroy the container.  Am I missing something?

HuKeping commented:
Maybe needs a re-trigger

mountkin commented:
The janky failure seems unrelated to this change.
You can run `git rebase -i HEAD~3` and change the `pick` in the 2nd and 3rd lines to `squash`, this way you can squash the 3 commits into one. After that run `git push -f origin 16756-dockercli-killtest` to force push the change to github.
Once the push finishes, janky will start the build automatically.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "16756-dockercli-killtest" git@github.com:zelahi/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~5
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

vdemeester commented:
@zelahi it seems there is some unrelated changes in this PR (`daemon.md` and `docker_cli_nat_test.go`), you're gonna need to remove these from the commit :wink:.

zelahi commented:
Ahh shoot, thanks for catching that @vdemeester . I can take care of it

LK4D4 committed:
Use more clear message for icc=false + iptables=false

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Merge pull request #17148 from albers/completion-dm-options

bash completion: support for more devicemapper options

Commit:
Merge pull request #17135 from albers/completion-restart-unless-stopped

Bash completion for restart policy `unless-stopped`

Commit:
Merge pull request #17186 from LK4D4/icc_clear

Use more clear message for icc=false + iptables=false

Commit:
Merge pull request #17187 from mavenugo/dopts

Fail the container start if the network has been removed

Commit:
Merge pull request #17088 from Microsoft/sjw/archive_hang_fix

Fixing hang in archive.CopyFileWithTar with invalid dst

thaJeztah committed:
Merge pull request #17094 from GabrielNicolasAvellaneda/docs-fixes

MINOR documentation fix.

zelahi commented:
Actually before I attempt something I do want to ask. Is there a recommended way to remove files that got into a commit?

HuKeping commented:
@zelahi since the branch for this PR is `16756-dockercli-killtest` and I think there must be a `master` branch and it should be update to the upstream/master. If so you can do
```
git checkout origin/master FILENAME
```

SvenDowideit committed:
Merge pull request #16801 from aidanhs/aphs-add-socat-t

Update ambassador image, use the socat -t option

zelahi committed:
Merge branch '16756-dockercli-killtest' of https://github.com/zelahi/docker into 16756-dockercli-killtest

Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "16756-dockercli-killtest" git@github.com:zelahi/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~14
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

zelahi committed:
Merge branch '16756-dockercli-killtest' of https://github.com/zelahi/docker into 16756-dockercli-killtest

Signed-off by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

vdemeester commented:
@zelahi Could you fix this PR or close it and re-open one with a clean commit please :wink:

zelahi commented:
Ahh sry, sure I can close this and open a clean pull request

vdemeester commented:
@zelahi Thank you very much :bow:
